### DISCUSSION TITLE
An updated client for mldonkey written with modern technologies (mldonkey-next)

### DISCUSSION THREAD

**Discussion by carlonluca** [ygrek/mldonkey#121] [General]
Hello! For those looking for a modern client for mldonkey, I wrote one for myself.

**mldonkey-next** is a a responsive Angular webapp, which can therefore be used from the browser. The webapp needs a bridge to connect to the core, so another process is needed. To simplify the installation, I maintain a docker app (**docker-mldonkey**) including the mldonkey core, mldonkey-next and the bridge.

I also created a native Qt app written in C++ embedding both the bridge and the webapp: this makes it possible to use this Qt app with any mldonkey installation. This app can currently be used for Linux as an AppImage or in Android.

For those interested, this is the official repo: https://github.com/carlonluca/mldonkey-next. For more info: https://bugfreeblog.duckdns.org/tag/mldonkey-next.

Contributions or bug reports are welcome.

**Comment by HinTak** [comment]
The first question is, can it load an existing mldonkey's config files and half-downloaded cache? It should be able to do it one-off (I.e. a one-way migration/conversion). If it cannot, you shouldn't use that name, because it is not next/upgrade, it is just another piece of software that serves a similar purpose. If it can, then you should say so, and quantify it - is it one-off conversion (i.e. once "next" loads and then save/append, it is no longer usable by non-next), completely compatible, or something between.

**Reply by carlonluca** [reply]
Hello! mldonkey-next is a client for mldonkey, not a replacement for the core. If you refer to docker-mldonkey instead, then yes, it uses an existing data dir of mldonkey, with its config file and cache, because it still is mldonkey from this repo (it may include additional patches, but strives to stay in sync with this repo by upstreaming those patches).

**Comment by HinTak** [comment]
Okay, in that case, it probably should be called "mldonkey useful addons" or "mldonkey extras"...

**Comment by carlonluca** [comment]
For those interested, I also implemented a Flatpak for mldonkey-next which should be a good alternative to the AppImage: https://flathub.org/en/apps/org.duckdns.bugfreeblog.mldonkeynext.
Regards.
